osquery/sql/virtual_table.cpp

```diff
@@ -74,6 +73,194 @@ static inline std::string opString(unsigned char op) {
   return "?";
 }
 
+/// A list of tables that come from extensions; it is used to determine which
+/// table can be read/write
+std::vector<std::string> extension_table_list;
+Mutex extension_table_list_mutex;
+
+bool isExtensionTable(const std::string& table_name) {
+  ReadLock lock(extension_table_list_mutex);
+  return (std::find(extension_table_list.begin(),
+                    extension_table_list.end(),
+                    table_name) != extension_table_list.end());
+}
+
+namespace {
+// A map containing an sqlite module object for each virtual table
+std::unordered_map<std::string, struct sqlite3_module> sqlite_module_map;
+Mutex sqlite_module_map_mutex;
+
+bool getColumnValue(std::string& value,
+                    int index,
+                    int argc,
```

Comment:
`index` and `argc` are used for indexing, would you like to define them as `std::size_t`? To avoid a problems with negative values.

Reply:
See https://github.com/facebook/osquery/pull/4094#discussion_r202101631